### DISCUSSION TITLE
Sidecar Support (lambda support)

### DISCUSSION THREAD

**Discussion by adrianspacely** [spatie/browsershot#519] [Ideas]
I'm currently looking into using [hammerstone/sidecar](https://github.com/hammerstonedev/sidecar) for PDF generation using puppeteer on an AWS node lambda.

Browsershot has a flexible enough API for me to piggyback off of it. My plan is to pass the lambda function the config, and a public single-use URL to the contents. My Laravel app will execute the lambda function via Sidecar, and the lambda function will return the PDF contents to my Laravel app.

Would Spatie consider adding sidecar support, or their own lambda execution support to this package? 🙏😁

_...the benefit, of course, being sidecar is so easy to use re: setting up lambdas; users of this package not requiring additional server software; and browsershot working in Laravel Vapor._

**Update:** I base64 encoded the HTML and the only update I did to the browsershot script was `await page.setContent(Buffer.from(event.html, 'base64').toString())`

**Comment by freekmurze** [comment]
We're not using Sidecar atm, so we're not going to invest time in it right now.

If the code to add this is simple enough, I'd accept a PR that adds support for Sidecar.
